onnxruntime/core/providers/cuda/test/beam_search_topk.cc

```diff
@@ -0,0 +1,135 @@
+#ifndef NDEBUG
```

Comment:
Do we need this ? Can we run the tests always ?

Reply:
yes, we need it. After cuda ep refactor, we can not access code in for cuda. Those tests are called when initializing cuda ep.